pkg/scalers/activemq_scaler.go

```diff
@@ -0,0 +1,282 @@
+	}
+
+	if meta.password == "" {
+		return nil, fmt.Errorf("password cannot be empty")
+	}
+
+	if val, ok := config.TriggerMetadata["metricName"]; ok {
+		meta.metricName = kedautil.NormalizeString(fmt.Sprintf("activemq-%s", val))
+	} else {
+		meta.metricName = kedautil.NormalizeString(fmt.Sprintf("activemq-%s", meta.destinationName))
+	}
+
+	meta.scalerIndex = config.ScalerIndex
+
+	return &meta, nil
+}
+
+func (s *activeMQScaler) IsActive(ctx context.Context) (bool, error) {
+	queueSize, err := s.getQueueMessageCount(ctx)
+	if err != nil {
+		activeMQLog.Error(err, "Unable to access activeMQ management endpoint", "managementEndpoint", s.metadata.managementEndpoint)
+		return false, err
+	}
+
+	return queueSize > 0, nil
+}
+
+// getRestAPIParameters parse restAPITemplate to provide managementEndpoint, brokerName, destinationName
+func getRestAPIParameters(meta activeMQMetadata) (activeMQMetadata, error) {
+	u, err := url.ParseRequestURI(meta.restAPITemplate)
+	if err != nil {
+		return meta, fmt.Errorf("unable to parse ActiveMQ restAPITemplate: %s", err)
+	}
+
+	meta.managementEndpoint = u.Host
+	splitURL := strings.Split(strings.Split(u.Path, ":")[1], "/")[0] // This returns : type=Broker,brokerName=<<brokerName>>,destinationType=Queue,destinationName=<<destinationName>>
+	replacer := strings.NewReplacer(",", "&")
+	v, err := url.ParseQuery(replacer.Replace(splitURL)) // This returns a map with key: string types and element type [] string. : map[brokerName:[<<brokerName>>] destinationName:[<<destinationName>>] destinationType:[Queue] type:[Broker]]
+	if err != nil {
+		return meta, fmt.Errorf("unable to parse ActiveMQ restAPITemplate: %s", err)
+	}
+
+	if len(v["destinationName"][0]) == 0 {
+		return meta, errors.New("no destinationName is given")
+	}
+	meta.destinationName = v["destinationName"][0]
+
+	if len(v["brokerName"][0]) == 0 {
+		return meta, fmt.Errorf("no brokerName given: %s", meta.restAPITemplate)
+	}
+	meta.brokerName = v["brokerName"][0]
+
+	return meta, nil
+}
+
+func (s *activeMQScaler) getMonitoringEndpoint() (string, error) {
+	var buf bytes.Buffer
+	endpoint := map[string]string{
+		"ManagementEndpoint": s.metadata.managementEndpoint,
+		"BrokerName":         s.metadata.brokerName,
+		"DestinationName":    s.metadata.destinationName,
+	}
+	template, err := template.New("monitoring_endpoint").Parse(defaultActiveMQRestAPITemplate)
+	if err != nil {
+		return "", fmt.Errorf("error parsing template: %s", err)
+	}
+	err = template.Execute(&buf, endpoint)
+	if err != nil {
+		return "", fmt.Errorf("error executing template: %s", err)
+	}
+	monitoringEndpoint := buf.String()
+	return monitoringEndpoint, nil
+}
+
+func (s *activeMQScaler) getQueueMessageCount(ctx context.Context) (int, error) {
+	var monitoringInfo *activeMQMonitoring
+	var queueMessageCount int
+
+	client := s.httpClient
+	url, err := s.getMonitoringEndpoint()
+	if err != nil {
+		return -1, err
+	}
+
+	req, err := http.NewRequestWithContext(ctx, "GET", url, nil)
```

Comment:
I think it would be better to check the error immediately after creating the request, because `req` might be returned as `nil` if err is not nil

```suggestion
	req, err := http.NewRequestWithContext(ctx, "GET", url, nil)
	if err != nil {
	    return -1, err
	}
```